File: day1/first/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	input, err := os.Open("../input.txt")
	if err != nil {
		os.Exit(1)
	}
	scanner := bufio.NewScanner(input)

	values := []int{}
	for scanner.Scan() {
		line := scanner.Text()
		var l, r rune
		for _, c := range line {
			if c >= '0' && c <= '9' {
				if l == 0 {
					l = c
				}
				r = c
			}
		}
		val, _ := strconv.Atoi(string(l) + string(r))
		values = append(values, val)
	}

	sum := 0
	for _, val := range values {
		sum += val
	}
	fmt.Println(sum)
}

File: day1/first/main.py
values = []

with open("../input.txt", "r") as file:
    while (line := file.readline()) and line:
        l, r = '', ''
        for c in line:
            if c.isnumeric():
                if l == '':
                    l, r = c, c
                else:
                    r = c
        values.append(int(l+r))

print(sum(values))